src/store/getDataSlice.js

```diff
@@ -0,0 +1,60 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { db } from "../firebase-config";
+import { collection, getDocs, addDoc } from "firebase/firestore";
+
+const dataInitialState = {
+  categoryLength: 4,
+  questionLength: 20,
+  update: false,
+  quizCollectionData: null,
+};
+
+const getDataSlice = createSlice({
+  name: "data",
+  initialState: dataInitialState,
+  reducers: {
+    updateQuizLength(state, action) {
+      if (
+        action.payload.cL !== state.categoryLength ||
+        action.payload.qL !== state.questionLength
+      ) {
+        state.categoryLength = action.payload.cL;
+        state.questionLength = action.payload.qL;
+        state.update = true;
+      } else {
+        state.update = false;
+      }
+    },
+    getDataFromFirebase(state, action) {
+      state.quizCollectionData = action.payload[0].dummyQuiz;
+      console.log("updated !", state.quizCollectionData);
+    },
+  },
+});
+
+export const sendData = (dummyQuiz, quizUpdated) => {
```

Comment:
Having difficulty in understanding the execution of this function. But if it is working it is good to go!!